### PR TITLE
fix(source): remove epoch in state user key and no flush on idle

### LINKED ISSUE #2546: source: avoid state explosion and unneccessary writes to state store
Currently, the number of kv paris writted and stored in state store keeps growing and the storage space will not be recalimed even though data ingestion has been stopped. After some diggings, I found out that there are several issues in our current source executor implementation:
- We encode epoch in the source state user key, which means compaction cannot expire states from previous epochs. However, we do not delete source state of old epochs explicitly so the source state keeps growing.
- We take a snapshot of the source state and write to storage on every barrier. We can save some writes if there is no data being ingested since the source state won't change in this case.

### PR DESCRIPTION
**Pull request by hzxa21**
## What's changed and what's your intention?
Fix #2546 

- Use state_identifier directly for source state key without encoding epoch in it.
- Add `idle_between_barriers` in source executor to track whether there is any data ingesting from source between barriers. Do not update source state to storage if the source state does not change.

With this PR, I confirmed (via grafana and manual checking on minio) that the number of newly added SSTs and number of kvs written to storage drop to zero after I stop producing data into the kafka source.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

### CODE REVIEW

**Review by skyzh — APPROVED** [review]

File: risedev.yml [hunk 1]
```diff
@@ -62,7 +62,9 @@ risedev:
   full:
     - use: minio
     - use: etcd
+    # - use: jaeger
```
Comment (lmatz) [inline risedev.yml:65]: typo?

File: src/connector/src/state.rs [hunk 2]
```diff
@@ -141,63 +92,29 @@ impl<S: StateStore> SourceStateHandler<S> {
...
-            }
-            Err(e) => Err(anyhow!(e)),
-        }
+    /// The function returns the serialized state.
+    pub async fn restore_states(
+        &self,
+        state_identifier: String,
+        epoch: u64,
+    ) -> Result<Option<Bytes>> {
+        self.keyspace.get(state_identifier, epoch).await.map_err(|e| anyhow!(e))
```
Comment (tabVersion) [inline src/connector/src/state.rs:101]: according to #2546,  we should not write a state if there is no change so we may not find a state for some epochs. I suggest returning the state with the first epoch smaller or equal to the given epoch.

File: src/stream/src/executor/source.rs [hunk 3]
```diff
@@ -61,6 +61,8 @@ pub struct SourceExecutor<S: StateStore> {
     split_state_store: SourceStateHandler<S>,
     // store latest split to offset mapping
     state_cache: Option<Vec<ConnectorState>>,
+    // whether there is any data flowing between two barriers
+    idle_between_barriers: bool,
```
Comment (tabVersion) [inline src/stream/src/executor/source.rs:65]: it's unnecessary to add a field here. we can make `state_cache = None` after flushing it to the state store. If `state_cache` is None, the flush can be skipped.

File: risedev.yml [hunk 4]
```diff
@@ -62,7 +62,9 @@ risedev:
...
     - use: meta-node
+      # enable-dashboard-v2: true
```
Comment (tabVersion) [inline risedev.yml:67]: plz remove unrelated changes

File: src/connector/src/state.rs (hunk 2 above)
Comment (skyzh) [reply src/connector/src/state.rs:101]: The `get` semantics is exactly "return the state with the first epoch smaller or equal to the given epoch".

File: src/stream/src/executor/source.rs (hunk 3 above)
Comment (hzxa21) [reply src/stream/src/executor/source.rs]: Good point. Done.

File: src/connector/src/state.rs [hunk 5]
```diff
@@ -141,63 +90,31 @@ impl<S: StateStore> SourceStateHandler<S> {
     /// Initializes the state of the specified ``state_identifier`` and returns an empty vec
```
Comment (lmatz) [inline src/connector/src/state.rs:90]: Seems the comment is outdated, no `vec` any more.

**Review by lmatz — APPROVED** [review]
